src/app.module.ts

```diff
@@ -1,12 +1,25 @@
 import { Module } from "@nestjs/common";
 import { AppController } from "./app.controller";
 import { AppService } from "./app.service";
-import { PostsController } from './posts/posts.controller';
-import { PostsService } from './posts/posts.service';
+import { TypeOrmModule } from "@nestjs/typeorm";
+import { Post } from "./posts/post.entity";
+import { PostsModule } from "./posts/posts.module";
 
 @Module({
-  imports: [],
-  controllers: [AppController, PostsController],
-  providers: [AppService, PostsService],
+  imports: [
+    TypeOrmModule.forRoot({
+      type: "mysql",
+      host: "localhost",
+      port: 3306,
+      username: "root",
+      password: "mysqlrootpassword",
+      database: "Blogtestdb",
+      entities: [Post],
+      synchronize: true,
```

Comment:
Synchronize is fine when developing but you need to take care since it might delete your data when you change entity on PROD 

What it does is whenever you change your entity it automatically syncs it with your database, and it might delete the table data just to be able to make these changes. we can use migrations instead.

![image](https://github.com/mohanad-80/nest-server/assets/87689547/24fa2a26-53c8-450c-95b0-c785dbe3948b)